### PR TITLE
Adds null handling in Power operator. (^)

### LINKED ISSUE #1297: "16^Blank()" and "Blank()^16" leads to a null exception
The ^ operator inserts a Call node to the Power function in IRTranslator.cs:
```
                switch (kind)
                {
                    // Call Node Replacements:
                    case BinaryOpKind.Power:
                        binaryOpResult = new CallNode(context.GetIRContext(node), BuiltinFunctionsCore.Power, left, right);
                        break;
```
And the Power entry in Library.cs says that handling blanks is handled by the IR:
```
            {
                BuiltinFunctionsCore.Power,
                StandardErrorHandling<NumberValue>(
                    BuiltinFunctionsCore.Power.Name,
                    expandArguments: NoArgExpansion,
                    replaceBlankValues: NoOpAlreadyHandledByIR,
                    checkRuntimeTypes: ExactValueTypeOrBlank<NumberValue>,
                    checkRuntimeValues: DeferRuntimeValueChecking,
                    returnBehavior: ReturnBehavior.AlwaysEvaluateAndReturnResult,
                    targetFunction: Power)
            },
```
And indeed, there is a pre-processor setup for Power in Power.cs:
```
        public override ArgPreprocessor GetArgPreprocessor(int index)
        {
            return base.GetGenericArgPreprocessor(index);
        }
```
However, since the Call node is inserted directly by the IR Translator, the standard function processing, like inserting blank checks into the IR, is skipped.

We should go through all the places we insert direct calls and ensure that the library replaces blanks rather than having the IR do it.  Or have the insert of Call nodes follow standard procedures for arg preprocessing.

### PR DESCRIPTION
**Pull request by jas-valgotar**
-Fixes #1297

### CODE REVIEW

**Comment by LucGenetier** [comment]
✅ No public API change.

**Review by CarlosFigueiraMSFT — APPROVED** [review]
🚢

**Review by CarlosFigueiraMSFT — DISMISSED** [review]
🚢

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/libraries/Microsoft.PowerFx.Core/IR/IRTranslator.cs [hunk 1]
```diff
@@ -208,8 +208,10 @@ public override IntermediateNode Visit(TexlBinaryOpNode node, IRTranslatorContex
                 switch (kind)
                 {
                     // Call Node Replacements:
-                    case BinaryOpKind.Power:
-                        binaryOpResult = new CallNode(context.GetIRContext(node), BuiltinFunctionsCore.Power, left, right);
+                    case BinaryOpKind.Power:
+                        var args = new List<IntermediateNode> { left, right };
+                        args = AttachArgPreprocessor(args, BuiltinFunctionsCore.Power, node, 2, context);
```
Comment (ian-legler) [inline src/libraries/Microsoft.PowerFx.Core/IR/IRTranslator.cs:213]: There should probably be some code comments here explaining why the IR is doing explicit arg preprocessing here and not in other call or binary op nodes generated

**Review by MikeStall — APPROVED** [review]
:shipit:

File: src/libraries/Microsoft.PowerFx.Core/IR/IRTranslator.cs (hunk 1 above)
Comment (jas-valgotar) [reply src/libraries/Microsoft.PowerFx.Core/IR/IRTranslator.cs:216]: Added. Thanks!

File: src/libraries/Microsoft.PowerFx.Core/Functions/TexlFunction.cs [hunk 2]
```diff
@@ -1462,9 +1462,9 @@ internal virtual IntermediateNode CreateIRCallNode(CallNode node, IRTranslatorCo
         /// By default, function does not attach any pre-processing for arguments.
         /// </summary>
         /// <param name="index">0 based index of argument.</param>
-        /// <param name="node">The original call node.</param>
+        /// <param name="argCount">The count of argument to call the function.</param>
```
Comment (CarlosFigueiraMSFT) [inline src/libraries/Microsoft.PowerFx.Core/Functions/TexlFunction.cs:1465]: >The count of argument to call the function. [](http://example.com/codeflow?start=35&length=43)

Can you give an example of this parameter? It's not clear based on the description.

**Review by MikeStall — DISMISSED** [review]

**Review by CarlosFigueiraMSFT — APPROVED** [review]
🚢

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/libraries/Microsoft.PowerFx.Core/Functions/TexlFunction.cs (hunk 2 above)
Comment (jas-valgotar) [reply src/libraries/Microsoft.PowerFx.Core/Functions/TexlFunction.cs]: Added more description.

**Review by CarlosFigueiraMSFT — APPROVED** [review]
🚢

**Comment by LucGenetier** [comment]
✅ No public API change.

**Pull request merged**